proxy/src/deshred.rs

```diff
@@ -434,6 +421,18 @@ fn update_state_tracker(shred: &Shred, state_tracker: &mut ShredsStateTracker) -
     }
     if let Shred::ShredData(s) = &shred {
         state_tracker.data_shreds[index] = Some(shred.clone());
+        if !s.last_in_slot() {
+            // If we received the next fec index before the current one, we could have perpetually
+            // failed to reconstruct the entries from the index due to never finding the start data boundary.
+            // To deal with this possible out-of-order scenario, we retry rebuilding the entrie(s) for the next
+            // fec index whenever we encounter a data complete boundary.
+            if let Some(next_slot) = &state_tracker.data_shreds[index + 1] {
```

Comment:
im a bit confused by this logic, `state_tracker.data_shreds` only tracks shreds for a single slot. how does this give you next slot information?

Reply:
also we only ever push to this when shreds come out of order. if they always come in order, the never push to the vec, and line 186 `    for (slot, fec_set_index) in data_fec_indexes_to_reconstruct.iter() {` wouldnt iterate on anything